### PR TITLE
Ignore nil input revisions

### PR DESCRIPTION
**Pull request by jveski**
It doesn't make sense to block synthesis when some inputs don't set a revision. If ordering isn't important, other controllers should be able to just omit the revisions.

### CODE REVIEW

File: internal/controllers/reconciliation/crud_test.go [hunk 1]
```diff
@@ -403,18 +403,7 @@ func TestReconcileInterval(t *testing.T) {
...
-	syn.Name = "test-syn"
-	syn.Spec.Image = "create"
-	require.NoError(t, upstream.Create(ctx, syn))
-
-	comp := &apiv1.Composition{}
-	comp.Name = "test-comp"
-	comp.Namespace = "default"
-	comp.Spec.Synthesizer.Name = syn.Name
-	require.NoError(t, upstream.Create(ctx, comp))
+	writeGenericComposition(t, upstream)
```
Comment (jveski) [inline internal/controllers/reconciliation/crud_test.go:406]: Bit of unrelated test refactoring

**Review by AYM1607 — APPROVED** [review]
